src/extension/logging.c

```diff
@@ -133,19 +139,25 @@ static void _ensure_init()
     if (!atomic_load_explicit(&_initialized, memory_order_acquire)) {
         TSRM_MUTEX_LOCK(_mutex);
         if (!atomic_load(&_initialized)) {
-            dd_result res = _do_dd_log_init();
-            if (res) {
+            dd_logging_init res = _do_dd_log_init();
+            if (res == logging_init_error) {
                 _mlog_php_varargs(
                     dd_log_warning, "Could not initialize logging");
                 _log_strategy = log_use_nothing;
             }
-            atomic_store_explicit(&_initialized, true, memory_order_release);
+            if (res != logging_init_try_later) {
```

Comment:
```suggestion
            else if (res != logging_init_try_later) {
```

Reply:
I don't think this is quite the same. If the response of `_do_dd_log_init` is not try later, the logging should be saved as initialized

Reply:
Ah right, you do want to "initialise" on error.